accessibility/demo/src/main/java/com/novoda/accessibility/demo/custom_actions/TweetView.java

```diff
@@ -0,0 +1,122 @@
+        super.onFinishInflate();
+        services = AccessibilityServices.newInstance(getContext());
+
+        View.inflate(getContext(), R.layout.merge_tweet, this);
+        tweetTextView = (TextView) findViewById(R.id.tweet_text);
+        replyButton = (Button) findViewById(R.id.tweet_button_reply);
+        retweetButton = (Button) findViewById(R.id.tweet_button_retweet);
+    }
+
+    public void display(final String tweet, final Listener listener) {
+        final Actions actions = createActions(tweet, listener);
+        ViewCompat.setAccessibilityDelegate(this, new ActionsAccessibilityDelegate(getResources(), actions));
+
+        tweetTextView.setText(tweet);
+
+        setOnClickListener(
+                new OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        if (services.isSpokenFeedbackEnabled()) {
+                            showAlertDialogFor(actions);
+                        } else {
+                            listener.onClick(tweet);
+                        }
+                    }
+                }
+        );
+
+        replyButton.setOnClickListener(
+                new OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        listener.onClickReply(tweet);
+                    }
+                }
+        );
+
+        retweetButton.setOnClickListener(
+                new OnClickListener() {
+                    @Override
+                    public void onClick(View v) {
+                        listener.onClickRetweet(tweet);
+                    }
+                }
+        );
+    }
+
+    private Actions createActions(final String tweet, final Listener listener) {
+        return new Actions(
+                Arrays.asList(
+                        new Action(R.id.tweet_action_open, R.string.tweet_action_open, new Runnable() {
+                            @Override
+                            public void run() {
+                                listener.onClick(tweet);
+                            }
+                        }),
+                        new Action(R.id.tweet_action_reply, R.string.tweet_action_reply, new Runnable() {
+                            @Override
+                            public void run() {
+                                listener.onClickReply(tweet);
+                            }
+                        }),
+                        new Action(R.id.tweet_action_retweet, R.string.tweet_action_retweet, new Runnable() {
+                            @Override
+                            public void run() {
+                                listener.onClickRetweet(tweet);
+                            }
+                        })
+                )
+        );
+    }
+
+    private void showAlertDialogFor(Actions actions) {
+        new ActionsAlertDialogCreator(getContext(), R.string.tweet_actions_title, actions)
```

Comment:
Would then need reference to FragmentManager.

The only benefit would be that it stays open when you rotate, and I don't think it should in this case:

- a11y focus doesn't persist on rotate (even without use of dialogs)
- this means if a dialog is open for a specific item, the device rotates, and the dialog is still open and the user presses back, then nothing will be focused (could expect the item to be focused)
- the above assumes that there is correct state persistence logic in the listview/viewpager to scroll to the same item over rotates. if not, it's even more confusing.

Are there other benefits to the DialogFragment apart from rotate?